testing/integration/upgrade_test.go

```diff
@@ -162,92 +143,92 @@ func upgradeMarkerRemoved() bool {
 }
 
 func TestStandaloneUpgrade(t *testing.T) {
-	info := define.Require(t, define.Requirements{
-		// Stack:   &define.Stack{},
+
+	define.Require(t, define.Requirements{
+		Stack:   &define.Stack{},
 		Local:   false, // requires Agent installation
 		Isolate: true,
 		Sudo:    true, // requires Agent installation
 	})
 
-	testSuite := &StandaloneUpgradeTestSuite{
-		requirementsInfo: info,
-		agentVersion:     define.Version(),
-	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	suite.Run(t, testSuite)
-}
+	versionList, latestVersion := getUpgradableAndLatestVersions(ctx, t)
 
-type StandaloneUpgradeTestSuite struct {
-	suite.Suite
+	// FIXME agent 7.X has a different control protocol which makes the upgrade, status command tricky (the fixture allocates a v2 client)
```

Comment:
Are you referring to the check where you call `c.State(ctx)` and compare the returned state with `cproto.State_HEALTHY`?  If so, could you instead use the CLI's `status` command with `--output json` and compare value of the `.Status` or `.state` fields to `cproto.State_HEALTHY`?

Reply:
I am referring to both checking the state and triggering the upgrade in the standalone case.
Sure I can use the CLI by parsing stdout, it's just that I was hoping not to...

Reply:
Using the CLI might help you avoid the issue with the different control protocols (the FIXME mentioned here).